test/pubspec_test.dart

```diff
@@ -443,17 +444,31 @@ dependencies:
     });
 
     group("environment", () {
+      /// Checking for the default sdk constraint based on the current sdk.
+      void expectDefaultSdkConstraint(Pubspec pubspec) {
+        var sdkVersionString = sdk.version.toString();
+        if (sdkVersionString.startsWith('2.0.0') && sdk.version.isPreRelease) {
```

Comment:
I'm not a fan of conditional test expectations.

Can you just override the SDK version number somehow to test it explicitly?

Reply:
with these tests I can't since they aren't spinning up a new process so I can't set an environment variable... I could more explicitly just do an `anyOf` check with the two allowed versions though?

I do have more explicit checks around this behavior in the version solver test (where I can set a fake sdk version), this is really just a way to get this test passing and confirming there is some sensible default constraint.